File: src/main/java/net/hectus/neobb/turn/default_game/flower/TWitherRose.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.SupernaturalClazz;
import org.bukkit.block.Block;

import java.util.List;

public class TWitherRose extends FlowerTurn {
    public TWitherRose(NeoPlayer player) { super(player); }
    public TWitherRose(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 5;
    }

    @Override
    public void apply() {
        player.game.allowedClazzes().remove(SupernaturalClazz.class);
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Luck(Buff.BuffTarget.NEXT, -15));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/flower/TWitherRose.kt
package net.hectus.neobb.modes.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.SupernaturalClazz
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TWitherRose(data: Block?, cord: Cord?, player: NeoPlayer?) : FlowerTurn(data, cord, player) {
    override val cost: Int = 5

    override fun apply() {
        player!!.game.allowed.remove(SupernaturalClazz::class)
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(Luck(-15, Buff.BuffTarget.NEXT))
    }
}
